multi-node/aws/cmd/kube-aws/command_up.go

```diff
@@ -116,14 +130,89 @@ func getCloudFormation(url string) (string, error) {
 
 	return string(tmpl), nil
 }
+func mustTarDirectory(basepath, path string) *bytes.Buffer {
+	buf := &bytes.Buffer{}
+
+	b64Writer := base64.NewEncoder(base64.StdEncoding, buf)
+	defer b64Writer.Close()
+
+	gzWriter, err := gzip.NewWriterLevel(b64Writer, gzip.BestCompression)
+	if err != nil {
+		stderr("Error creating gzip writer: %s", err)
+		os.Exit(1)
+	}
+	defer gzWriter.Close()
+
+	tarWriter := tar.NewWriter(gzWriter)
+	defer tarWriter.Close()
+
+	tarHandler := func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			//Terminate on error
+			return err
+		}
+		hdr, err := tar.FileInfoHeader(info, "")
+		if err != nil {
+			return err
+		}
+		relPath, err := filepath.Rel(basepath, path)
+		if err != nil {
+			return err
+		}
+
+		hdr.Name = relPath
+
+		if err = tarWriter.WriteHeader(hdr); err != nil {
+			return err
+		}
+
+		if !info.IsDir() {
+			f, err := os.Open(path)
+			if err != nil {
+				return err
+			}
+			defer f.Close()
+			if _, err := io.Copy(tarWriter, f); err != nil {
+				return err
+			}
+		}
+
+		return nil
+	}
+
+	if err := filepath.Walk(path, tarHandler); err != nil {
+		stderr("Error tar-ing directory %s: %s", path, err)
+		os.Exit(1)
+	}
+	return buf
+}
+
+func mustReadFile(loc string) *bytes.Buffer {
```

Comment:
This function name doesn't accurately describe what this function does anymore (read file / gzip / base64 encode)